### PR TITLE
Statistics: simplify CPU recorder when workload on a thread can always be attributed to a request

### PR DESCRIPTION
**Pull request by zhongzc**
### What problem does this PR solve?

Problem Summary:

The previous CPU sampling way takes the CPU time spent in the subsequent 10ms of the sampling point as the CPU time of requests.

This complex approach comes from a focus on performance and accuracy.

First of all, it is quite expensive to read the thread stat, so we should try not to do it when there is no request load on the thread. However, In case a thread has no request loads for a long time and suddenly needs to process a request, the interval between two reads of stat will be long. If the thread handles a lot of non-request load during the interval, then that occasional request will incorrectly record too much CPU time.

The previous solution is to reduce the interval between two reads of the stat. The implementation is to read stat at two consecutive sampling points only after the request load appears.

On the other hand, there is another simpler way to implement via eliminating non-request loads. That's why this PR is proposed.

### What is changed and how it works?

Traverse the code to ensure that there are no non-request loads running on the unified read pool.

What's Changed:

- Wrap all read requests of storage with `in_resource_metering_tag`.
- Record CPU time in an intuitive way.

It should be noted that the previously recorded CPU time doesn't include runtime overhead, whereas now it does. Some people will prefer the current implementation, after all, runtime overhead is also brought by the request.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by zhongzc** [comment]
@crazycs520 PTAL

**Review by crazycs520 — COMMENTED** [review]
Miss add `in_resource_metering_tag` for `batch_get_command` ?

File: components/resource_metering/src/cpu/recorder/linux.rs [hunk 1]
```diff
@@ -248,41 +245,30 @@ impl CpuRecorder {
...
-                            .starts_with(super::TEST_TAG_PREFIX)),
-                        |_| {}
-                    );
-
-                    // Store the beginning stat for the current tag.
-                    if cur_tag.is_some() {
-                        thread_stat.prev_tag = cur_tag;
-                        thread_stat.prev_stat = stat;
+                    let prev_cpu_ticks =
+                        (thread_stat.stat.utime as u64).wrapping_add(thread_stat.stat.stime as u64);
```
[the remark this answers is not in the document]
Comment (zhongzc) [reply components/resource_metering/src/cpu/recorder/linux.rs]: Done

File: src/storage/mod.rs [hunk 2]
```diff
@@ -418,7 +418,9 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         // all requests in a batch have the same region, epoch, term, replica_read
         let priority = requests[0].get_context().get_priority();
         let concurrency_manager = self.concurrency_manager.clone();
-        let resource_tag = ResourceMeteringTag::from_rpc_context(requests[0].get_context());
+        let rand_index = rand::thread_rng().gen_range(0, requests.len());
```
Comment (breeswish) [inline src/storage/mod.rs:421]: Prefer to add a comment so that others will know why do we need a random.

**Review by breeswish — APPROVED** [review]
the rest LGTM

**Review by breeswish — APPROVED** [review]

**Comment by breeswish** [comment]
@crazycs520 PTAL

**Review by andylokandy — APPROVED** [review]

**Pull request merged**